app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/miband/MiBand2Support.java

```diff
@@ -104,12 +108,25 @@ public void onReceive(Context context, Intent intent) {
 
             } else if (s.equals(BatteryInfoProfile.ACTION_BATTERY_INFO)) {
                 handleBatteryInfo((nodomain.freeyourgadget.gadgetbridge.service.btle.profiles.battery.BatteryInfo) intent.getParcelableExtra(BatteryInfoProfile.EXTRA_BATTERY_INFO));
+            } else if (s.equals(DeviceService.ACTION_MIBAND2_AUTH)) {
+                byte[] response = intent.getExtras().getByteArray(DeviceService.EXTRA_MIBAND2_AUTH_BYTE);
+                BluetoothGattCharacteristic temp = getCharacteristic(MiBand2Service.UUID_UNKNOWN_CHARACTERISTIC9);
+                temp.setValue(response);
+                if ((temp.getProperties() & (BluetoothGattCharacteristic.PROPERTY_WRITE | BluetoothGattCharacteristic.PROPERTY_WRITE_NO_RESPONSE)) != 0) {
+                    Log.d("HELLO", "its looks ok");
+                }
+                if (!mBluetoothGatt.writeCharacteristic(temp)) {
+                    Log.d("ERROR", "ERROR");
+                }
+                Log.d("HELLO", "HELLO");
+
             }
         }
     };
 
     private volatile boolean telephoneRinging;
     private volatile boolean isLocatingDevice;
+    private BluetoothGatt mBluetoothGatt;
```

Comment:
Is it necessary that you remember the BluetoothGatt here and directly write to it? I'd prefer if you could use the TransactionBuilder /BtleQueue to do this.

If this doesn't work for some reason, I'd like to figure out why it doesn't work and if we could improve the infrastructure to make it work. 

But in the meantime, feel free to keep it that way so that you can progress without waiting for better infrastructure.